crates/adapters/src/catalog.rs

```diff
@@ -1,20 +1,142 @@
-use crate::{DeCollectionHandle, DeZSetHandle, SerOutputBatchHandle};
-use dbsp::{algebra::ZRingValue, CollectionHandle, DBData, DBWeight};
-use serde::Deserialize;
+use crate::{
+    DeCollectionHandle, DeScalarHandle, DeScalarHandleImpl, DeZSetHandle, SerOutputBatchHandle,
+};
+use dbsp::{
+    algebra::ZRingValue,
+    operator::{DelayedFeedback, NeighborhoodDescr},
+    CollectionHandle, InputHandle, RootCircuit, Stream, ZSet,
+};
+use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
+use utoipa::ToSchema;
+
+// This is only here so we can derive `ToSchema` for it without adding
+// a `utoipa` dependency to the `dbsp` crate to derive ToSchema for
+// `NeighborhoodDescr`.
+/// A request to output a specific neighborhood of a table or view.
+/// The neighborhood is defined in terms of its central point (`anchor`)
+/// and the number of rows preceding and following the anchor to output.
+#[derive(Deserialize, ToSchema)]
+pub struct NeighborhoodQuery {
+    pub anchor: utoipa::openapi::Object,
```

Comment:
so you have utoipa here but don't need a depedency?

Reply:
This is `adapters` crate.  I don't want a `utoipa` dependency in `dbsp`.